### PR TITLE
docs: create JS functions documentation

### PR DESCRIPTION
**Pull request by jacekroszkowiakdev**
The documentation includes information about functions and scope, lexical scope, and closures to provide more detailed insight into how functions work and behave in JavaScript.

### CODE REVIEW

File: Functions/introduction_to_functions_and_scope.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+Variables declared in a function are in local scope. They are not accessible outside of the function.
+
+```js
+var fn = function () {
+    var n = 0;
+    console.log(typeof n);
+};
+
+fn(); //logs 'number'
+console.log(typeof n);
```
Comment (artrugh) [inline Functions/introduction_to_functions_and_scope.md]: I guess it is missing the: //logs undefined

File: Functions/scope_lexical_scope_and_closures.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            console.log(this.name); // 'Alice'
+        }, 1000);
+    },
+};
+
+obj.traditionalFunc(); // Executes with traditional function
+obj.arrowFunc(); // Executes with arrow function
+```
+
+## Closures
```
Comment (artrugh) [inline Functions/scope_lexical_scope_and_closures.md:93]: could u write some example of a closure please?

File: Functions/introduction_to_functions_and_scope.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+}
+```
+
+Function **parameters** are listed inside the parentheses () in the function definition.
+
+Function **arguments** are the values received by the function when it is invoked.
+
+Inside the function, the arguments (the parameters) behave as local variables.
+
+![JavaScript Function visualized](./images/JavaScript_functions_visualized.png)
```
Comment (artrugh) [inline Functions/introduction_to_functions_and_scope.md]: name of files, snake case!

Comment (jacekroszkowiakdev) [reply Functions/introduction_to_functions_and_scope.md]: my bad, thanks for spotting this.

File: Functions/introduction_to_functions_and_scope.md (hunk 1 above)
Comment (jacekroszkowiakdev) [reply Functions/introduction_to_functions_and_scope.md]: Exactly, missed that while copying from dillinger.io

File: Functions/scope_lexical_scope_and_closures.md (hunk 2 above)
Comment (jacekroszkowiakdev) [reply Functions/scope_lexical_scope_and_closures.md:93]: Sure!

**Comment by artrugh** [comment]
Check again the naming pather for files and folders please

**Pull request closed without merging**